### PR TITLE
Add Smoke Tests

### LINKED ISSUE #83: Incorporate Smoke Tests for Build and Packaging into the Jenkins build pipelines
Now we have tests in https://github.com/AdoptOpenJDK/openjdk-build/tree/master/test/functional/buildAndPackage, we should incorporate them into the Jenkins build pipelines.

They ideally should be run prior to runTests and should block the running of other tests (currently there is an outstanding failure in the smoke tests that should get fixed as we will block all testing if failures occur in the [SmokeTestsFoBuildAndPackaging job](https://ci.adoptopenjdk.net/view/work-in-progress/job/SmokeTestsForBuildandPackaging/), related: https://github.com/AdoptOpenJDK/openjdk-build/issues/1387).  Additional context described here: AdoptOpenJDK/openjdk-tests#2024.

This will require use of some extra test job parameters:
  - VENDOR_TEST_REPOS=https://github.com/AdoptOpenJDK/openjdk-build
  - VENDOR_TEST_BRANCHES=master
  - VENDOR_TEST_DIRS=/test/functional

### PR DESCRIPTION
**Pull request by sophia-guo**
Close #83 

Signed-off-by: Sophia Guo <sophia.gwf@gmail.com>

### CODE REVIEW

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
run tests

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (M-Davies) [inline pipelines/build/common/openjdk_build_pipeline.groovy:142]: These may not always use `ADOPT_DEFAULTS_JSON`. If this is being executed from a user's repository and perspective, this will pull in Adopt's vendor details.

Could we have a check to see if the user's respective `DEFAULTS_JSON` values match Adopt's and, if they don't, the user's DEFAULTS_JSON parameters should be used instead? Let me know if you need further explanation of what I mean 🙂

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (M-Davies) [inline pipelines/build/common/openjdk_build_pipeline.groovy:303]: Unneeded comment

File: pipelines/defaults.json
Comment (M-Davies) [inline pipelines/defaults.json:5]: Will need to update the `UsingOurScripts.md` documentation to explain this addition

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (M-Davies) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1385]: Little confused what this for? Why not miss this out completely?

**Review by M-Davies — CHANGES_REQUESTED** [review]

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
run tests

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (sophia-guo) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: Thanks @M-Davies ! Updated with using `buildConfig.USE_ADOPT_SHELL_SCRIPTS`, which I think should handle the case of executing from a user's perspective.

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
run tests

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (M-Davies) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: Yep looks good! Thanks

**Comment by M-Davies** [comment]
@sophia-guo Is this ready for review now?

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
Depends on https://github.com/AdoptOpenJDK/openjdk-tests/pull/2364

**Comment by sophia-guo** [comment]
@M-Davies it's ready for review. Thanks.

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (M-Davies) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1400]: Do we always want to have smoke tests enabled? For those who may not want to run said tests because they only want to run a quick test build or have no need to run these tests (i.e. third parties who wish to build using our scripts) then this is not ideal. I would prefer to have it inside the `enableTests && buildConfig.TEST_LIST.size() > 0` conditional.

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (M-Davies) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1407]: Indentation

**Review by M-Davies — CHANGES_REQUESTED** [review]

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (aahlenst) [reply pipelines/build/common/openjdk_build_pipeline.groovy:1400]: If those tests fail, the build is in shambles. `openjdk-build` didn't do what it was supposed to do. That's the reason those tests are in `openjdk-build` and not over in `openjdk-tests`. The number of tests is small and a test run is quick (less than a minute). So there should not be a reason not to run these tests and it's the reason why they are a hard failure contrary to our normal tests that are soft.

There could be an escape hatch, but it may not be `buildConfig.TEST_LIST.size()` (or anything else used to normally skip the tests) if it comes from this line:

https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/blob/b50b6ab0344d2c55c19863aca3c6e22e23f2db22/pipelines/jobs/configurations/jdk17_pipeline_config.groovy#L7

Because we have specialists that disable those tests, although, in that case, the build should not have been enabled in the first place.

**Comment by sophia-guo** [comment]
@M-Davies it's ready for review but I would like to hold it before https://ci.adoptopenjdk.net/view/work-in-progress/job/SmokeTestsForBuildandPackaging/ is good, which relates with AdoptOpenJDK/openjdk-build#1387.  Except from convert it to draft do you have other methods to block it from merge?

**Comment by M-Davies** [comment]
> @M-Davies it's ready for review but I would like to hold it before https://ci.adoptopenjdk.net/view/work-in-progress/job/SmokeTestsForBuildandPackaging/ is good, which relates with [AdoptOpenJDK/openjdk-build#1387](https://github.com/AdoptOpenJDK/openjdk-build/issues/1387). Except from convert it to draft do you have other methods to block it from merge?

We are on build lockdown anyway due to the JDK16 release so I'll add that label and see what the state of this PR is afterwards

**Comment by karianna** [comment]
@M-Davies we're unlocked after 16 now so you can go ahead.

**Review by M-Davies — CHANGES_REQUESTED** [review]
Requires rebase due to https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/pull/102

**Comment by smlambert** [comment]
Since https://github.com/AdoptOpenJDK/openjdk-build/issues/1387 is now merged, this one can be rebased and merged.

**Review by smlambert — APPROVED** [review]

**Comment by smlambert** [comment]
@M-Davies - are you good to proceed with this PR?  It shows you as requesting a change, have all your requests been met?

**Comment by M-Davies** [comment]
> @M-Davies - are you good to proceed with this PR? It shows you as requesting a change, have all your requests been met?

Still looking for some clarification for [this](https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/pull/86#discussion_r592178628)

**Comment by smlambert** [comment]
IMHO, enableTests could just as well be called enableAQATests.  The smoke tests are different, they are not part of the AQA tests, they are checking the tarball was built with the correct features enabled, and correct version information and metadata has been be produced.  There could then be a enableSmokeTests option, because one may want smoke tests but not AQA tests... 

I don't want us to get bogged down in this however, it is more important that we are running these then whether some person kicking off build pipelines is monkeying with settings.  

So the proposal to 
```
if (enableTests) {
    runSmokeTests()
    if (buildConfig.TEST_LIST.size() > 0) {
        try {
```
is fine by me.

**Comment by sophia-guo** [comment]
@M-Davies updated.

**Comment by M-Davies** [comment]
run tests

**Review by M-Davies — APPROVED** [review]
PR tester is down due to the current jenkins instability. There's nothing I can see in the code that would indicate it would fail so I'm happy to approve this and submit a test pipeline after merge

**Comment by smlambert** [comment]
If you are ready to merge this @M-Davies please do so (I do not believe @sophia-guo has write permissions to this repo, so she would be unable to merge her own).

**Comment by M-Davies** [comment]
I believe @andrew-m-leonard is doing the m2 builds at the moment so I'll merge tomorrow morning when they are done

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (M-Davies) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: Resolved by https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/pull/86#issuecomment-812107292

**Comment by gdams** [comment]
run tests

**Review by gdams — APPROVED** [review]

**Comment by gdams** [comment]
run tests

**Comment by gdams** [comment]
run tests

**Comment by M-Davies** [comment]
I would merge but it appears I do not have write access to this repo anymore since it was changed to adoptium :(

**Comment by smlambert** [comment]
@M-Davies - you are listed as a committer (https://projects.eclipse.org/proposals/eclipse-temurin), so please check your Eclipse profile (and make sure that your github ID is listed in it... you can see it in social media section when you click Edit profile).

**Comment by gdams** [comment]
Looks like it's there: https://projects.eclipse.org/user/15355

That is rather strange?

**Comment by M-Davies** [comment]
I'll try and sort it out on slack to avoid the spam here

**Comment by smlambert** [comment]
I've sent a note to emo team (& cc-ed @M-Davies ) - I think what happened is that Morgan was on the committer list for the [Temurin project proposal](https://projects.eclipse.org/proposals/eclipse-temurin), but not in the list for the [actual project](https://projects.eclipse.org/projects/adoptium.temurin/who) once the proposal was approved.  Vague recollection of Wayne not receiving Morgan's email in time for project creation.  It can and should be added now, and that is what is requested by the email to emo team.

**Pull request merged**
